pallets/liquid-staking-v2/src/lib.rs

```diff
@@ -92,8 +106,140 @@ mod pallet {
         EraAlreadyPushed,
     }
 
+    #[pallet::genesis_config]
+    pub struct GenesisConfig {
+        pub exchange_rate: Rate,
+    }
+
+    #[cfg(feature = "std")]
+    impl Default for GenesisConfig {
+        fn default() -> Self {
+            Self {
+                exchange_rate: Rate::default(),
+            }
+        }
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config> GenesisBuild<T> for GenesisConfig {
+        fn build(&self) {
+            ExchangeRate::<T>::put(self.exchange_rate);
+        }
+    }
+
+    #[cfg(feature = "std")]
+    impl GenesisConfig {
+        /// Direct implementation of `GenesisBuild::build_storage`.
+        ///
+        /// Kept in order not to break dependency.
+        pub fn build_storage<T: Config>(&self) -> Result<sp_runtime::Storage, String> {
+            <Self as GenesisBuild<T>>::build_storage(self)
+        }
+
+        /// Direct implementation of `GenesisBuild::assimilate_storage`.
+        ///
+        /// Kept in order not to break dependency.
+        pub fn assimilate_storage<T: Config>(
+            &self,
+            storage: &mut sp_runtime::Storage,
+        ) -> Result<(), String> {
+            <Self as GenesisBuild<T>>::assimilate_storage(self, storage)
+        }
+    }
+
     #[pallet::call]
     impl<T: Config> Pallet<T> {
+        /// Put assets under staking, the native assets will be transferred to the account
+        /// owned by the pallet, user receive derivative in return, such derivative can be
+        /// further used as collateral for lending.
+        ///
+        /// - `amount`: the amount of staking assets
+        #[pallet::weight(T::WeightInfo::stake())]
+        #[transactional]
+        pub fn stake(origin: OriginFor<T>, amount: Balance) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+
+            let exchange_rate = ExchangeRate::<T>::get();
+            let derivative_amount = exchange_rate
```

Comment:
I think we are mixing two words here:

liquid & derivative

maybe we can choose one and keep it consistent

Reply:
changed to liquid_amount